### DISCUSSION TITLE
How do I reference a file in a chat with an agent which uses the OpenAI Assistants API (and the SK Experimental Agents feature)

### DISCUSSION THREAD

**Discussion by Ryan-Britton** [microsoft/semantic-kernel#5103] [Q&A] [answered]
I am experimenting with the Experimental Agents package - and can't seem to figure out how to upload a file in a particular user message. In my use case my assistant should read and answer questions on an uploaded file in a user message (not seeded into OpenAI storage for Retrieval, preferably). How would I go about doing that? Even if I *do* upload the file into OpenAI storage for retrieval - how do I reference the file id in my user message?

basically mimicking this:

![image](https://github.com/microsoft/semantic-kernel/assets/124330161/975f8839-5ee9-4a98-a338-448cc593b170)

**Comment by rogerbarreto** [comment, marked as answer]
@crickman, Worth taking a look on this, looking at the `ChatThread` class it seems to be using only the `Id` from ThreadModel, might worth allowing in a ChatThread to reference file ids or an abstraction for Files.

**Comment by 3052** [comment]
I have reposted this question because it seems @sophialagerkranspandey incorrectly closed this

https://github.com/microsoft/semantic-kernel/discussions/12714
